crates/autopilot/src/periodic_db_cleanup.rs

```diff
@@ -0,0 +1,134 @@
+use {
+    crate::database::Postgres,
+    chrono::{DateTime, Utc},
+    std::time::Duration,
+    tokio::time,
+};
+
+pub struct OrderEventsCleanerConfig {
+    cleanup_interval: Duration,
+    event_age_threshold: chrono::Duration,
+}
+
+impl OrderEventsCleanerConfig {
+    pub fn new(cleanup_interval: Duration, event_age_threshold: Duration) -> Self {
+        OrderEventsCleanerConfig {
+            cleanup_interval,
+            event_age_threshold: chrono::Duration::from_std(event_age_threshold).unwrap(),
+        }
+    }
+}
+
+pub struct OrderEventsCleaner {
+    config: OrderEventsCleanerConfig,
+    db: Postgres,
+}
+
+impl OrderEventsCleaner {
+    pub fn new(config: OrderEventsCleanerConfig, db: Postgres) -> Self {
+        OrderEventsCleaner { config, db }
+    }
+
+    pub async fn run_forever(self) -> ! {
+        let mut interval = time::interval(self.config.cleanup_interval);
+        loop {
+            let timestamp: DateTime<Utc> = Utc::now() - self.config.event_age_threshold;
+            match self.db.delete_events_before(timestamp).await {
+                Ok(affected_rows_count) => {
+                    tracing::debug!(
+                        "deleted {:?} order events before {}",
+                        affected_rows_count,
+                        timestamp
+                    );
```

Comment:
nit: We try to have logs which have a static string and log any additional data as named variables like this:
```rust
tracing::debug!(affected_rows, timestamp, "deleted old order events");
```
This generally makes it nicer to find these logs in kibana because their string is always constant while still being able to circle in on logs with specific values when that's needed.

Reply:
Updated the log